src/tokio/bufread/generic/decoder.rs

```diff
@@ -162,3 +162,33 @@ impl<R: AsyncBufRead, D: Decode> AsyncRead for Decoder<R, D> {
         }
     }
 }
+
+impl<R: AsyncBufRead + AsyncWrite, D: Decode> AsyncWrite for Decoder<R, D> {
```

Comment:
Thanks now I understand it use case and I think it fits in to the abstraction layer.

Can you also add this to the `tokio::bufread::Encoder`?

And `future::bufread::*coder` if possible, I prefer the API to be consistent across future/tokio, decode/encode.

If you think it's too difficult, then we can do it in a separate PR.